MNE/disp3D/3DObjects/bem/bemsurfacetreeitem.cpp

```diff
@@ -97,6 +97,8 @@ BemSurfaceTreeItem::BemSurfaceTreeItem(int iType, const QString& text)
 
 BemSurfaceTreeItem::~BemSurfaceTreeItem()
 {
+    setVisible(false);
+    delete m_pRenderable3DEntity;
```

Comment:
m_pRenderable3DEntity is an QEntity, which inherits from QObject. However, since m_pRenderable3DEntity's parent is set to m_pParentEntity, m_pParentEntity will delete all its child automatically. Therefore deleting m_pRenderable3DEntity here should also be unnecessary. In this case using a QPointer would function as an extra safeguard mechanism but would also not actually be needed? Pls correct if I amwrong.